### PR TITLE
Update images to 2.10.1 and fix errors

### PR DESCRIPTION
**Pull request by foxzi**
Update docker files centos and ubuntu for fixing error and update to 2.10 version
    
    In centos docker file fixed problem with gosu for arm64 platform, update gosu version.
    Remove excess variable GC64 which used only for centos and ubuntu images and didn't used
    in .gtilab.mk, add using common variable LUAJIT_ENABLE_GC64.

### CODE REVIEW

File: versions/ubuntu_20.04_2.10.1 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+2
+
```
Comment (ylobankov) [inline versions/ubuntu_20.04_2.10.1:2]: 1. Please remove empty line.
2. What about 2.10.0?

File: versions/ubuntu_20.04_1.10.14 [hunk 2]
```diff
@@ -0,0 +1,2 @@
+1.10
+
```
Comment (ylobankov) [inline versions/ubuntu_20.04_1.10.14:2]: 1. Please remove empty line.

Comment (foxzi) [reply versions/ubuntu_20.04_1.10.14:2]: done

File: versions/ubuntu_20.04_2.10.1 (hunk 1 above)
Comment (foxzi) [reply versions/ubuntu_20.04_2.10.1:2]: 1. done
2. I don't think that it's need to change, because it past.

File: dockerfiles/centos_7 [hunk 3]
```diff
@@ -1,12 +1,10 @@
-ARG BASE_IMAGE
-FROM ${BASE_IMAGE}centos:7
+FROM centos:7
 LABEL org.opencontainers.image.authors="artembo@me.com; piligrim@rootnix.net"
 
 ARG TNT_VER \
-    GC64
```
Comment (NickVolynkin) [inline dockerfiles/centos_7:6]: What's the poing of replacing GC64 with LUAJIT_ENABLE_GC64 and then using LUAJIT_ENABLE_GC64 to set GC64?

File: versions/centos_7_2.10.1 [hunk 4]
```diff
@@ -0,0 +1 @@
+2
```
Comment (NickVolynkin) [inline versions/centos_7_2.10.1:1]: These files should have exact versions like `2.10.1-1-<commit-sha>`

Comment (NickVolynkin) [reply versions/centos_7_2.10.1:1]: - When releasing 2.10.1 we should also update files *.2.10 and *.2
- If files *.2.10.0 are missing, we should add them too.

File: dockerfiles/ubuntu_20.04 [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
+FROM ubuntu:20.04
 LABEL org.opencontainers.image.authors="artembo@me.com; piligrim@rootnix.net"
 
 ARG TNT_VER \
-    GC64
+    LUAJIT_ENABLE_GC64
 
-ENV GC64=${GC64:-false} \
-    TARANTOOL_URL=https://www.tarantool.io/release/${TNT_VER:-2}/installer.sh \
+ENV TARANTOOL_URL=https://www.tarantool.io/release/${TNT_VER}/installer.sh \
```
Comment (NickVolynkin) [inline dockerfiles/ubuntu_20.04:7]: Why? https://www.tarantool.io/release/2.10.0/installer.sh returns 404

File: dockerfiles/centos_7 [hunk 6]
```diff
@@ -1,12 +1,10 @@
-ARG BASE_IMAGE
-FROM ${BASE_IMAGE}centos:7
+FROM centos:7
```
Comment (NickVolynkin) [inline dockerfiles/centos_7:1]: Why did you replace it? To make Dockerfile usable without makefile?

This variable is set in https://github.com/tarantool/docker/blob/master/.gitlab.mk#L18. If we don't need it, it should be removed from Makefile as well.

And it had the `${ARCH}` part, where did it go?

File: dockerfiles/centos_7 [hunk 7]
```diff
@@ -20,7 +18,7 @@ ENV GC64=${GC64:-false} \
     LUAROCK_TARANTOOL_GIS_VERSION=1.0.1 \
     LUAROCK_TARANTOOL_PROMETHEUS_VERSION=1.0.4 \
     LUAROCK_TARANTOOL_GPERFTOOLS_VERSION=1.0.1 \
-    GOSU_VERSION=1.11
+    GOSU_VERSION=1.14
```
Comment (NickVolynkin) [inline dockerfiles/centos_7:21]: Did you just update it because there's a new version? Was there a bug?

File: dockerfiles/centos_7 [hunk 8]
```diff
@@ -97,9 +96,11 @@ repo_gpgcheck = 1\n'\
         perl \
     && rm -rf /var/cache/yum \
     && : "---------- gosu ----------" \
+    && if [[ $(uname -m) = 'x86_64' ]]; then GOSU_ARCH='amd64'; fi \
```
Comment (NickVolynkin) [inline dockerfiles/centos_7:99]: This part looks great. How did it work before?

**Review by NickVolynkin — CHANGES_REQUESTED** [review]

**Comment by NickVolynkin** [comment]
Please update the version table in the README https://github.com/tarantool/docker/blob/master/README.md#build-pipelines

File: versions/ubuntu_20.04_2.10.1 (hunk 1 above)
Comment (NickVolynkin) [reply versions/ubuntu_20.04_2.10.1]: 2. We still can rebuild images of 2.10.0, for example, with new rocks versions.

File: versions/centos_7_2.10.1 (hunk 4 above)
Comment (NickVolynkin) [reply versions/centos_7_2.10.1:1]: For example, see https://github.com/tarantool/docker/commit/d9de887713ccdd89d6aeb9b94dd83c75d10b572c

File: dockerfiles/centos_7 (hunk 8 above)
Comment (foxzi) [reply dockerfiles/centos_7:99]: It didn't work before, I really don't know how was built images before with this bag

File: dockerfiles/centos_7 (hunk 6 above)
Comment (foxzi) [reply dockerfiles/centos_7:1]: You are right. Variable ARCH, BASE_IMAGE etc we don't use. Ok I will remove it.

File: dockerfiles/centos_7 (hunk 3 above)
Comment (foxzi) [reply dockerfiles/centos_7:6]: Variable LUAJIT_ENABLE_GC64 set before this modification and functionality GC64 variable duplicates it.

File: .gitlab.mk [hunk 9]
```diff
@@ -13,7 +13,7 @@ build:
 		--build-arg ENABLE_BUNDLED_LIBYAML=${ENABLE_BUNDLED_LIBYAML} \
 		--build-arg TNT_VER=${TNT_VER} \
 		--build-arg LUAJIT_DISABLE_SYSPROF=${LUAJIT_DISABLE_SYSPROF} \
-		--build-arg LUAJIT_ENABLE_GC64=${LUAJIT_ENABLE_GC64} \
+		--build-arg LUAJIT_ENABLE_GC64=${GC64} \
```
Comment (ylobankov) [inline .gitlab.mk:16]: Looks wrong, AFAIU `LUAJIT_ENABLE_GC64` -> `GC64`.

File: dockerfiles/centos_7 [hunk 10]
```diff
@@ -36,6 +36,7 @@ repo_gpgcheck = 1\n'\
 >> /etc/yum.repos.d/pg.repo \
     && groupadd tarantool \
     && adduser -g tarantool tarantool \
+    && if [[ $LUAJIT_ENABLE_GC64 -eq 'OFF' ]]; then export GC64=false; else export GC64=true; fi\
```
Comment (ylobankov) [inline dockerfiles/centos_7]: Looks like it shouldn't be added.

File: dockerfiles/ubuntu_20.04 [hunk 11]
```diff
@@ -38,6 +38,7 @@ RUN set -x \
     ca-certificates \
     && ln -snf /usr/share/zoneinfo/UTC /etc/localtime && echo UTC > /etc/timezone \
     && ln -s /usr/lib/$(arch)-linux-gnu/libprofiler.so.0.4.18 /usr/lib/$(arch)-linux-gnu/libprofiler.so \
+    && if [[ $LUAJIT_ENABLE_GC64 -eq 'OFF' ]]; then export GC64=false; else export GC64=true; fi\
```
Comment (ylobankov) [inline dockerfiles/ubuntu_20.04]: ditto

File: .gitlab.mk (hunk 9 above)
Comment (foxzi) [reply .gitlab.mk]: done

File: dockerfiles/centos_7 (hunk 10 above)
Comment (foxzi) [reply dockerfiles/centos_7]: done

File: dockerfiles/ubuntu_20.04 (hunk 11 above)
Comment (foxzi) [reply dockerfiles/ubuntu_20.04]: fixed

File: .gitlab.mk [hunk 12]
```diff
@@ -3,7 +3,7 @@ ROCKS_INSTALLER?='tarantoolctl rocks'
 ENABLE_BUNDLED_LIBYAML?='ON'
 IMAGE?=tarantool/tarantool
 LUAJIT_DISABLE_SYSPROF?=OFF
-LUAJIT_ENABLE_GC64?=OFF
+GC64?=OFF
```
[the remark this answers is not in the document]
Comment (foxzi) [reply .gitlab.mk:6]: I think We shouldn't spend time to it.

**Pull request closed without merging**